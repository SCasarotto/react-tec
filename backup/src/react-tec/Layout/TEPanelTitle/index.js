import TEPanelTitle from './TEPanelTitle'

export default TEPanelTitle
