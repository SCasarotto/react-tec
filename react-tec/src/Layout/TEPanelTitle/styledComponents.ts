import styled from 'styled-components'

export const Container = styled.div`
	width: 100%;
	background-color: ${(props) => props.theme.white};
	display: flex;
	align-items: center;
	justify-content: space-between;
	padding-bottom: 10px;
	margin-bottom: 20px;
	border-bottom: 1px solid ${(props) => props.theme.lightGray};
`
export const Title = styled.h1`
	text-align: left;
	font-size: 30px;
	margin-top: 0px;
	margin-bottom: 0px;
	color: ${(props) => props.theme.primary};
	line-height: 1;

	@media (max-width: 1000px) {
		font-size: 28px;
	}
	@media (max-width: 800px) {
		font-size: 24px;
	}
	@media (max-width: 650px) {
		font-size: 20px;
	}
`
export const ComponentWrapper = styled.div`
	width: auto;
`
export const Subtitle = styled.p`
	font-size: 18px;
	margin-bottom: 0px;
	margin-top: 0px;
`
