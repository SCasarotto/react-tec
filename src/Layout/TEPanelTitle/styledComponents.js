import styled from 'styled-components'

export const Container = styled.div`
    ${(props) => {
        const { theme } = props
        return `
            width: 100%;
            backgroundColor: ${theme.white};
            display: flex;
            align-items: center;
            justify-content: space-between;
            padding-bottom: 10px;
            margin-bottom: 20px;
            border-bottom: 1px solid ${theme.lightGray};
        `
    }}
`
export const Title = styled.h1`
    ${(props) => {
        const { theme } = props
        return `
            text-align: left;
            font-size: 30px;
            margin-top: 0px;
            margin-bottom: 0px;
            color: ${theme.primary};
            line-height: 1;

            @media (max-width: 1000px) {
                font-size: 28px;
            }
            @media (max-width: 800px) {
                font-size: 24px;
            }
            @media (max-width: 650px) {
                font-size: 20px;
            }
        `
    }}
`
export const ComponentWrapper = styled.div`
    ${(props) => {
        return `
            width: auto;
        `
    }}
`
export const Subtitle = styled.p`
    ${(props) => {
        return `
            font-size: 18px;
            margin-bottom: 0px;
            margin-top: 0px;
        `
    }}
`
