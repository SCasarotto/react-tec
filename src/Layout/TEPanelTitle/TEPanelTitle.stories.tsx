import React from 'react'
import { Story, Meta } from '@storybook/react'

import { TEPanelTitle, TEPanelTitleProps } from './TEPanelTitle'

export default {
	title: 'Layout/TEPanelTitle',
	component: TEPanelTitle,
	argTypes: {},
} as Meta

const Template: Story<TEPanelTitleProps> = (args) => <TEPanelTitle {...args} />

export const Default = Template.bind({})
Default.args = {
	title: 'Title',
	subtitle: 'Subtitle',
}

export const Components = Template.bind({})
Components.args = {
	leftComponent: <button>Some Button</button>,
	rightComponent: <input />,
}
