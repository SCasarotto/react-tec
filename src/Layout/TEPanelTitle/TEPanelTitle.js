import React from 'react'
import PropTypes from 'prop-types'

import { Container, Title, ComponentWrapper, Subtitle } from './styledComponents'

const TEPanelTitle = (props) => {
	const { className = '', title, leftComponent, subtitle, rightComponent } = props

	return (
		<Container className={`TEPanelTitle ${className}`}>
			{title && <Title className='TEPanelTitleTitle'>{title}</Title>}
			{leftComponent && (
				<ComponentWrapper className='TEPanelTitleLeftComponentWrapper'>
					{leftComponent}
				</ComponentWrapper>
			)}
			{subtitle && <Subtitle className='TEPanelTitleSubTitle'>{subtitle}</Subtitle>}
			{rightComponent && (
				<ComponentWrapper className='TEPanelTitleRightComponentWrapper'>
					{rightComponent}
				</ComponentWrapper>
			)}
		</Container>
	)
}

TEPanelTitle.propTypes = {
	title: PropTypes.string,
	leftComponent: PropTypes.node,
	subtitle: PropTypes.string,
	rightComponent: PropTypes.node,
}

export default TEPanelTitle
