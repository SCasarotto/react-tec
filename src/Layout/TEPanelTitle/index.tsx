import React from 'react'

import { Container, Title, ComponentWrapper, Subtitle } from './styledComponents'

export interface TEPanelTitleProps {
	className?: string
	title?: string
	leftComponent?: React.ReactNode
	subtitle?: string
	rightComponent?: React.ReactNode
}
export const TEPanelTitle: React.FC<TEPanelTitleProps> = (props) => {
	const { className = '', title, leftComponent, subtitle, rightComponent } = props

	return (
		<Container className={`TEPanelTitle ${className}`}>
			{title && <Title className='TEPanelTitleTitle'>{title}</Title>}
			{leftComponent && (
				<ComponentWrapper className='TEPanelTitleLeftComponentWrapper'>
					{leftComponent}
				</ComponentWrapper>
			)}
			{subtitle && <Subtitle className='TEPanelTitleSubTitle'>{subtitle}</Subtitle>}
			{rightComponent && (
				<ComponentWrapper className='TEPanelTitleRightComponentWrapper'>
					{rightComponent}
				</ComponentWrapper>
			)}
		</Container>
	)
}
