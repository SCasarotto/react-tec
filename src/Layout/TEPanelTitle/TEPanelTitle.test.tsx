import React from 'react'
import { render } from '@testing-library/react'
import { TEPanelTitle } from './TEPanelTitle'

describe('TEPanelTitle', () => {
	// Renders Without Error
	it('matches snapshot', () => {
		const { asFragment } = render(<TEPanelTitle />)
		expect(asFragment()).toMatchSnapshot()
	})

	// Class Name
	it('accepts custom className', () => {
		const { container } = render(<TEPanelTitle className='myClassName' />)
		expect(container.firstChild).toHaveClass('myClassName')
	})
})
