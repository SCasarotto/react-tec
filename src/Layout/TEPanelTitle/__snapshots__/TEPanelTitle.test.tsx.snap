// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TEPanelTitle matches snapshot 1`] = `
<DocumentFragment>
  .c0 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding-bottom: 10px;
  margin-bottom: 20px;
  border-bottom: 1px solid;
}

<div
    class="c0 TEPanelTitle "
  />
</DocumentFragment>
`;
